lib/registry.go

```diff
@@ -16,55 +16,72 @@ import (
 
 // RegistryClientMethods defines business logic for working with registries
 type RegistryClientMethods struct {
-	inst *Instance
+	d dispatcher
 }
 
-// NewRegistryClientMethods creates client methods from an instance
-func NewRegistryClientMethods(inst *Instance) *RegistryClientMethods {
-	return &RegistryClientMethods{
-		inst: inst,
-	}
+// Name returns the name of this method group
+func (m RegistryClientMethods) Name() string {
+	return "registry"
 }
 
-// CoreRequestsName implements the Requests interface
-func (RegistryClientMethods) CoreRequestsName() string { return "registry" }
+// Attributes defines attributes for each method
+func (m RegistryClientMethods) Attributes() map[string]AttributeSet {
+	return map[string]AttributeSet{
+		"createprofile":   {denyRPC, ""},
+		"proveprofilekey": {denyRPC, ""},
+	}
+}
 
 // RegistryProfile is a user profile as stored on a registry
 type RegistryProfile = registry.Profile
 
+// RegistryProfileParams encapsulates arguments for creating or proving a registry profile
+type RegistryProfileParams struct {
+	Profile *RegistryProfile
+}
+
```

Comment:
again, `RegistryProfileParams` added b/c using a `RegistryProfile` triggers `Validate()` when dispatching, and this profile won't have every field filled in.